contractcourt/contract_resolvers.go

```diff
@@ -81,8 +70,15 @@ type ResolverKit struct {
 	// should write the state of the resolver to persistent storage, and
 	// return a non-nil error upon success.
 	Checkpoint func(ContractResolver) error
+}
+
+// contractResolverKit is meant to be used as a mix-in struct to be embedded within a
+// given ContractResolver implementation. It contains all the common items that
+// a resolver requires to carry out its duties.
+type contractResolverKit struct {
+	ResolverKit
 
-	Quit chan struct{}
+	quit chan struct{}
```

Comment:
make quit channel on the contract resolver instead?

Reply:
or maybe the code duplication gets worse... Just a bit weird with a wrapper just to add a quit channel

Reply:
i felt the same, couldn't come up with something better

Reply:
maybe this will be useful later if we get another shared field